test/raft/distributed_test.cc

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2021 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <seastar/testing/thread_test_case.hh>
+
+template <typename M> concept PureStateMachine =
+requires (typename M::state_t s, typename M::input_t i) {
+    typename M::state_t;
+    typename M::input_t;
+    typename M::output_t;
+    { M::delta(s, i) } -> std::same_as<std::pair<typename M::state_t, typename M::output_t>>;
+    M::init;
+    requires std::is_same_v<const typename M::state_t, decltype(M::init)>;
+};
+
+SEASTAR_THREAD_TEST_CASE(dummy_test) {
```

Comment:
Did you mean it to be here?

Reply:
I included it so I don't get linker errors (I think it would complain about lack of main function).

Reply:
Ah, I see. I guess another way to go about it is to not add it to compilation targets just yet, but this is fine as well.